bitnami/redis/templates/_helpers.tpl

```diff
@@ -82,25 +82,58 @@ Return the appropriate apiVersion for PodSecurityPolicy.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return true if a TLS secret object should be created
+*/}}
+{{- define "redis.createTlsSecret" -}}
+{{- if and .Values.tls.enabled .Values.tls.autoGenerated }}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the secret containing Redis TLS certificates
+*/}}
+{{- define "redis.tlsSecretName" -}}
+{{- $secretName := coalesce .Values.tls.existingSecret .Values.tls.certificatesSecret -}}
+{{- if $secretName -}}
+    {{- printf "%s" (tpl $secretName $) -}}
+{{- else -}}
+    {{- printf "%s-crt" (include "common.names.fullname" .) -}}
```

Comment:
I am not very sure when this would happen.

Reply:
If `.Values.tls.existingSecret` or `.Values.tls.certificatesSecret` are provided, it will be used as the TLS secret by default. Having `existingSecret` priority over `certificatesSecret` because of the [coalesce](https://helm.sh/docs/chart_template_guide/function_list/#coalesce).

If no existing secret is provided, then it will use `$fullname-crt`, which is the name of the auto-generated secret (Check `tls-secret.yaml`)